src/components/VirtualTypingKeyboard.vue

```diff
@@ -174,240 +172,78 @@ function windowKeyUp(event: KeyboardEvent) {
     shiftRightFill.value = 'lightgray'
   }
 }
-
-// Synth interactive state and properties
-
-const noteOffs: Map<number, NoteOff> = new Map()
-
-const isMousePressed = ref(false)
-
-function start(index: number) {
-  noteOffs.set(index, props.noteOn(index))
-}
-
-function end(index: number) {
-  if (noteOffs.has(index)) {
-    noteOffs.get(index)!()
-    noteOffs.delete(index)
-  }
-}
-
-function onTouchEnd(event: TouchEvent, index: number) {
-  event.preventDefault()
-  end(index)
-}
-
-function onTouchStart(event: TouchEvent, index: number) {
-  event.preventDefault()
-  // Make sure that we start a new note.
-  end(index)
-
-  start(index)
-}
-
-function onMouseDown(event: MouseEvent, index: number) {
-  if (event.button !== LEFT_MOUSE_BTN) {
-    return
-  }
-  event.preventDefault()
-  isMousePressed.value = true
-  start(index)
-}
-
-function onMouseUp(event: MouseEvent, index: number) {
-  if (event.button !== LEFT_MOUSE_BTN) {
-    return
-  }
-  event.preventDefault()
-  isMousePressed.value = false
-  end(index)
-}
-
-function onMouseEnter(event: MouseEvent, index: number) {
-  if (!isMousePressed.value) {
-    return
-  }
-  event.preventDefault()
-  start(index)
-}
-
-function onMouseLeave(event: MouseEvent, index: number) {
-  if (!isMousePressed.value) {
-    return
-  }
-  event.preventDefault()
-  end(index)
-}
-
-function windowMouseUp(event: MouseEvent) {
-  if (event.button === LEFT_MOUSE_BTN) {
-    isMousePressed.value = false
-  }
-}
+const playNote = usePlayNote(props.noteOn);
 
 onMounted(() => {
   props.typingKeyboard.addKeydownListener(typingKeyDown)
-  window.addEventListener('keydown', windowKeyDown)
-  window.addEventListener('keyup', windowKeyUp)
-  window.addEventListener('mouseup', windowMouseUp)
 })
 
 onUnmounted(() => {
   props.typingKeyboard.removeEventListener(typingKeyDown)
   window.removeEventListener('keydown', windowKeyDown)
   window.removeEventListener('keyup', windowKeyUp)
-  noteOffs.forEach((off) => {
-    if (off !== null) {
-      off()
-    }
-  })
-  window.removeEventListener('mouseup', windowMouseUp)
+  playNote.onUnmounted();
 })
 </script>
 
 <template>
   <svg width="200" height="100" viewBox="0 0 1700 800" style="width: 100%; height: auto">
     <g class="click-not-implemented">
-      <rect
-        x="100"
-        y="100"
-        width="98"
-        height="98"
-        rx="5"
-        ry="5"
-        :fill="noteOffFill"
-        stroke="lightslategray"
-        stroke-width="1"
-      />
-      <rect
-        x="125"
-        y="140"
-        width="50"
-        height="20"
-        fill="none"
-        stroke="dimgray"
-        stroke-width="1"
-        pointer-events="none"
-      />
+      <rect x="100" y="100" width="98" height="98" rx="5" ry="5" :fill="noteOffFill" stroke="lightslategray"
```

Comment:
While stuff like this arguably looks better, we use automatic formatting so just run `npm format` please. If you strongly disagree with the formatter make a separate PR tweaking its settings.